### ISSUE TITLE
The stopping condition of PARAFAC2 uses an absolute loss decrease condition, not a relative loss decrease condition

### ISSUE THREAD

**Issue by MarieRoald** [tensorly/tensorly#263]
In TensorLy, the PARAFAC2-ALS algorithm stops based on an absolute loss decrease condition. That is

```python
if rec_error[-2] - rec_error[-1] < tol:
    break
```

where `rec_error` is the Frobenius norm of the reconstruction error. However, the PARAFAC2-ALS algorithm is usually described with a relative loss decrease condition based on the SSE. That is

```python
if rec_error[-2]**2 - rec_error[-1]**2 < tol * rec_error[-2]**2:
    break
```

While this difference is not large, I believe it is better to follow the [reference MATLAB implementation](http://www.models.life.ku.dk/algorithms) by Rasmus Bro to prevent confusion. 

Unfortunately, this change will alter the behaviour for existing code. There are three options that I see:

- Make the change, with a note in the documentation
- Make the change, with a note in the documentation and a disableable warning
- Make the change, with an option to enable the old behaviour.
- Don't make any change

I don't know which is best, but I have implemented the first option and written documentation for it (in [this branch](https://github.com/MarieRoald/tensorly/tree/parafac2_stopping_condition)). If we decide to go with the first or second option, then I think we should specify which version the behaviour changed (like scikit-learn does, see for example the `validate` argument of the [`FunctionTransformer` documentation](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.FunctionTransformer.html)). Since I didn't know the relevant version, I left it as with a TODO in the docstring.

What do you think is best action here?

**Comment by JeanKossaifi** [comment]
Thanks @MarieRoald. In general I think we should go with the best possible default so happy to go with 1 (using the reference by Rasmus Bro + note in the documentation). Is there any study of the behaviour of various convergence criteria? 

We probably need to give some more thoughts to the convergence criteria in general - @cohenjer might have more experience with these.

**Comment by cohenjer** [comment]
Relative decrease is indeed generally better than absolute decrease. Absolute decrease is extremely problem-dependent while the relative decrease is a little more robust and a default value of say 1e-8 makes more sense.

Since the P2 in Tensorly is still relatively new, its Ok I think to go with 1 like Jean said.

**Comment by JeanKossaifi** [comment]
Fixed in #267, thanks @MarieRoald!
